### PR TITLE
Initial structure for offline memory screen.

### PR DESCRIPTION
**Pull request by polina-c**
Lint errors are fixed here: https://github.com/flutter/devtools/pull/5966

### CODE REVIEW

File: packages/devtools_app/lib/src/app.dart [hunk 1]
```diff
@@ -290,6 +291,17 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
           ),
         );
       },
+      memoryAnalysisScreenId: (_, __, args, ____) {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/app.dart:294]: Our standard navigation only works if there is live connection. We do not have mechanism to navigate to the same screen with connection and without it. If I pass `ScreenMetaData.memory.id` as route on button click at landing page, navigation does not happen.

So, with current mechanism we need different route to connect to the screen offline. 

I suggest to review/refactor this separately, in a right way, when we implement offline pattern.

[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/app.dart:295]: > Our standard navigation to a screen only works if there is live connection.

Not necessarily. We already can navigate to the app size screen and any standalone screens without an app connection. I'd take another look at the stack trace - my guess is that the live connection requirement is coming from the memory screen code itself.

[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/app.dart:295]: Fine to refactor later though. Please hide this route behind the `FeatureFlags.memoryAnalysis` flag. We don't want a user to be able to navigate to a screen that is not shipped.

[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/app.dart:295]: The code is copy-pasted from what we do for app size one block before:

```
appSizeScreenId: (_, __, args, ____) {
        final embed = isEmbedded(args);
        return DevToolsScaffold.withChild(
          key: const Key('appsize'),
          embed: embed,
          child: MultiProvider(
            providers: _providedControllers(),
            child: const AppSizeBody(),
          ),
        );
      },
```

[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/app.dart:295]: > The code is copy-pasted from what we do for app size one block before:

I'm not following. How does this relate to navigation? Navigation to the app size screen (which does not require an app connection) happens here: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/framework/landing_screen.dart#L351

I am saying that if you were unable to navigate to the memory screen without having an app connection, then that likely is due to code from the memory screen that assumes we have a connected app - not code from the navigation framework, which has no understanding of app connected state.

[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/app.dart:295]: Discussed offline. Fine to leave as is for now - we will likely refactor how both the app size and memory screens are accessed for static mode here soon with the evolution of static tooling support in DevTools.

File: packages/devtools_app/lib/src/app.dart [hunk 2]
```diff
@@ -290,6 +292,17 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
...
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/app.dart:295]:
```suggestion
  if (FeatureFlags.memoryAnalysis)
      memoryAnalysisScreenId: (_, __, args, ____) {
```

**Review by kenzieschmoll — CHANGES_REQUESTED** [review]
please hide behind the feature flag, then lgtm

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**